src/sbvr-api/permissions.ts

```diff
@@ -1002,16 +1039,54 @@ const getBoundConstrainedMemoizer = memoizeWeak(
 						// If the table is definition based then just make the bypass version match but pointing to the equivalent bypassed resources
 						constrainedAbstractSqlModel.tables[bypassResourceName].definition =
 							createBypassDefinition(table.definition);
+
+						// Make clear to TS that table.definition isn't going to become undefined by the time we use it inside the onceGetter below.
+						const tableDefinition = table.definition;
+
+						// For the transalated resources, we create a fake `resource$permissions"<permJson>"` resource
+						// and replace their nested Resource references to also point to that parallel `$permissions"<permJson>"` suffixed model
+						// so that once the whole chain of `resource$permissions"<permJson>"` -> `resource$vX$permissions"<permJson>"`-> `resource$vX+1$permissions"<permJson>"` Resource references
+						// is fully resolved, the permissions are applied to the last model in the chain.
+						// The permissions are added to the last model in the chain eg `resource$vDB$permissions"<permJson>"`,
+						// by the generateConstrainedAbstractSql() that's in the `onceGetter(permissionsTable, 'definition'` below.
+						for (const stringifiedPermission of _.uniq(
+							Object.values(stringifiedMethodPermissions),
+						)) {
+							if (stringifiedPermission === stringifiedMethodPermissions.GET) {
+								// The read permissions (for GET requests) are resolved using the plain resources in the constrainedAbstractSqlModel
+								// (those without a `$permissions` suffix, which get generated by the memoizedGetConstrainedModel),
+								// so we don't need to re-create a set of `<resource>$permissions"read"` resources.
+								continue;
+							}
+							// If the permission is not a GET then we need to create a permissions table
+							const permissionResourceName = `${resourceName}$permissions${stringifiedPermission}`;
+							constrainedAbstractSqlModel.tables[permissionResourceName] = {
+								...table,
+								resourceName: permissionResourceName,
+							};
+							onceGetter(
+								constrainedAbstractSqlModel.tables[permissionResourceName],
+								'definition',
+								() =>
+									createVersionSpecificPermissionDefinition(
+										tableDefinition,
+										stringifiedPermission,
+									),
+							);
+						}
 					} else {
 						// Otherwise constrain the non-bypass table
+						// When the table doesn't have a definition, then it's the last translation layer,
+						// in which case we define it as the virtual `resource$vDB$permissions"read"` resource,
+						// which invoikes the generateConstrainedAbstractSql() that's in the `onceGetter(permissionsTable, 'definition'` below
```

Comment:
```suggestion
						// which invokes the generateConstrainedAbstractSql() that's in the `onceGetter(permissionsTable, 'definition'` below
```

Reply:
I actually ended up rephrasing this a bit